src/viam/robot/client.py

```diff
@@ -753,6 +755,28 @@ async def stop_all(self, extra: Dict[ResourceName, Dict[str, Any]] = {}):
         request = StopAllRequest(extra=ep)
         await self._client.StopAll(request)
 
+    #######
+    # LOG #
+    #######
+
+    async def log(self, name: str, level: str, time, log: str, stack: str):
+        """Send log from Python module over gRPC.
+
+        Create a LogEntry object from the log to send to RDK.
+
+        Args:
+            name (str): The logger's name.
+            level (str): The level of the log.
+            time (str): The log creation time.
+            log (str): The log message.
+            caller (Dict[str, int]): The filename and number when the log was created.
+            stack_info (str): The stack information of the log.
+            fields (Dict[str, ValueTypes]): The extra fields of the log
+        """
+        entry = LogEntry(level=level, time=datetime_to_timestamp(time), logger_name=name, message=log, stack=stack)
```

Comment:
This is almost perfect; I worry about the value of `logger_name`. 

Right now, for a log from `my_gizmo.py`, `logger_name` here ends up being `src.gizmo.my_gizmo`. The eventual RDK log looks like: `2024-03-10T22:26:43.009-0400	INFO	robot_server.src.gizmo.my_gizmo`. As you can see, we use the `logger_name` on the RDK side as a suffix to `robot_server` to give the user an idea of where the log came from. In Go, I used the name of the module; that would be something like `"complex_module"` in this case. Is there any way to set `logger_name` to the name of the module here? That way, we could get an eventual RDK log that looks like: `2024-03-10T22:26:43.009-0400	INFO	robot_server.complex_module`.

Reply:
I'll try!

Reply:
Modules don't have `name` in Python. How do you do it in Go so that the module and/or loggers know the module name?

Reply:
Discussed offline; @dgottlieb how do you feel about Python module logs showing up as `2024-03-10T22:26:43.009-0400	INFO	robot_server.src.gizmo.my_gizmo`? Modules aren't really aware of their own name. In Go, we knew the module name because the module author _named_ the logger and we encouraged them to use the name of their module.

Reply:
I agree this is an interesting quirk. I'm fine with how this is right now.

If you _really_ wanted to try something (that may have different negative consequences), you could try trimming `src.gizmo.my_gizmo` down to just the bit after the last dot, i.e: `my_gizmo`.

Reply:
Agree with @dgottlieb (and I think you suggested that option offline to me already @purplenicole730); if it's non-trivial to trim then I think what you have now should work!

Reply:
Modules now look like this:
```
2024-03-12T12:59:16.238-0400	INFO	robot_server.my_gizmo	my_gizmo.py:79	gizmo1 is closed.
```